Update cargo

weihanglo opened the pull request:
9 commits in e027c4b5d25af2119b1956fac42863b9b3242744..f6e737b1e3386adb89333bf06a01f68a91ac5306
2022-11-25 19:44:46 +0000 to 2022-12-02 20:21:24 +0000
- Refactor generate_targets into separate module (https://github.com/rust-lang/cargo/pull/11445)
- Improve file found in multiple build targets warning (https://github.com/rust-lang/cargo/pull/11299)
- Error when precise without -p flag (https://github.com/rust-lang/cargo/pull/11349)
- Improve strategy for selecting targets to be scraped for examples (https://github.com/rust-lang/cargo/pull/11430)
- Aware of compression ratio for unpack size limit (https://github.com/rust-lang/cargo/pull/11337)
- Add test for rustdoc-map generation when using sparse registries (https://github.com/rust-lang/cargo/pull/11403)
- Add error message when `cargo fix` on an empty repo (https://github.com/rust-lang/cargo/pull/11400)
- Store the sparse+ prefix in the SourceId for sparse registries (https://github.com/rust-lang/cargo/pull/11387)
- Update documentation for -Zrustdoc-scrape-examples in the Cargo Book (https://github.com/rust-lang/cargo/pull/11425)

weihanglo commented:
@bors r+ p=1

bors commented:
:pushpin: Commit 1b366a645afc3301ab426496ffa1e77ac5e928c5 has been approved by `weihanglo`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=weihanglo 1b366a645afc3301ab426496ffa1e77ac5e928c5 -->
<!-- homu: {"type":"Approved","sha":"1b366a645afc3301ab426496ffa1e77ac5e928c5","approver":"weihanglo","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 1b366a645afc3301ab426496ffa1e77ac5e928c5 with merge a8ad52ae4e4d7c2031b79903bfc4d216ada7861b...
<!-- homu: {"type":"BuildStarted","head_sha":"1b366a645afc3301ab426496ffa1e77ac5e928c5","merge_sha":"a8ad52ae4e4d7c2031b79903bfc4d216ada7861b"} -->

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/3580137997/jobs/6022278085)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/3580137997/jobs/6022278085","builder_name":"checks-actions"} -->

rust-log-analyzer commented:
The job **`dist-i586-gnu-i586-i686-musl`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/3580137997/jobs/6021996121) [(plain)](https://github.com/rust-lang-ci/rust/commit/a8ad52ae4e4d7c2031b79903bfc4d216ada7861b/checks/9787226389/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
test str::char_count::zh_medium::case01_filter_count_cont_bytes    ... ok
test str::char_count::zh_medium::case02_iter_increment             ... ok
test str::char_count::zh_medium::case03_manual_char_len            ... ok
test str::char_count::zh_small::case00_libcore                     ... ok
error: test failed, to rerun pass `-p core --bench corebenches`
Caused by:
Caused by:
  process didn't exit successfully: `/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-std/i686-unknown-linux-musl/release/deps/corebenches-84f3cd4479fdb461` (signal: 11, SIGSEGV: invalid memory reference)
```

</details>

weihanglo committed:
Update cargo

9 commits in e027c4b5d25af2119b1956fac42863b9b3242744..f6e737b1e3386adb89333bf06a01f68a91ac5306
2022-11-25 19:44:46 +0000 to 2022-12-02 20:21:24 +0000
- Refactor generate_targets into separate module (rust-lang/cargo#11445)
- Improve file found in multiple build targets warning (rust-lang/cargo#11299)
- Error when precise without -p flag (rust-lang/cargo#11349)
- Improve strategy for selecting targets to be scraped for examples (rust-lang/cargo#11430)
- Aware of compression ratio for unpack size limit (rust-lang/cargo#11337)
- Add test for rustdoc-map generation when using sparse registries (rust-lang/cargo#11403)
- Add error message when `cargo fix` on an empty repo (rust-lang/cargo#11400)
- Store the sparse+ prefix in the SourceId for sparse registries (rust-lang/cargo#11387)
- Update documentation for -Zrustdoc-scrape-examples in the Cargo Book (rust-lang/cargo#11425)

weihanglo commented:
@bors r+ p=1

bors commented:
:pushpin: Commit 9bd6062833d2574866ed1c4804cb01e2e2e1afdf has been approved by `weihanglo`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=weihanglo 9bd6062833d2574866ed1c4804cb01e2e2e1afdf -->
<!-- homu: {"type":"Approved","sha":"9bd6062833d2574866ed1c4804cb01e2e2e1afdf","approver":"weihanglo","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 9bd6062833d2574866ed1c4804cb01e2e2e1afdf with merge dd7a12593a5274dd812a7b8ab91a4ff29adb81a2...
<!-- homu: {"type":"BuildStarted","head_sha":"9bd6062833d2574866ed1c4804cb01e2e2e1afdf","merge_sha":"dd7a12593a5274dd812a7b8ab91a4ff29adb81a2"} -->

rust-log-analyzer commented:
The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/3606283070/jobs/6077473466) [(plain)](https://github.com/rust-lang/rust/commit/9bd6062833d2574866ed1c4804cb01e2e2e1afdf/checks/9859327506/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
.......... (90/91)
.          (91/91)


/checkout/src/test/rustdoc-gui/basic-code.goml basic-code... FAILED
[ERROR] (line 3) expected 1 elements, found 0: for command `assert-count: (".src-line-numbers", 1)`
Build completed unsuccessfully in 0:02:28
```

</details>

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/3606513018/jobs/6079002547)
Approved by: weihanglo
Pushing dd7a12593a5274dd812a7b8ab91a4ff29adb81a2 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"weihanglo","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/3606513018/jobs/6079002547"},"merge_sha":"dd7a12593a5274dd812a7b8ab91a4ff29adb81a2"} -->

rust-timer commented:
Finished benchmarking commit (dd7a12593a5274dd812a7b8ab91a4ff29adb81a2): [comparison URL](https://perf.rust-lang.org/compare.html?start=e526d123aa4703aa7b85c5b8e70c20b5ee0deade&end=dd7a12593a5274dd812a7b8ab91a4ff29adb81a2&stat=instructions:u).

### Overall result: no relevant changes - no action needed


@rustbot label: -perf-regression




**Warning ⚠**: The following benchmark(s) failed to build:
- rustc


cc @rust-lang/wg-compiler-performance

### [Instruction count](https://perf.rust-lang.org/compare.html?start=e526d123aa4703aa7b85c5b8e70c20b5ee0deade&end=dd7a12593a5274dd812a7b8ab91a4ff29adb81a2&stat=instructions:u)
This benchmark run did not return any relevant results for this metric.

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=e526d123aa4703aa7b85c5b8e70c20b5ee0deade&end=dd7a12593a5274dd812a7b8ab91a4ff29adb81a2&stat=max-rss)
<details>
<summary>Results</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | 2.0%  | [1.8%, 2.4%]   | 3     |
| Improvements ✅ <br /> (primary)   | -1.5% | [-1.5%, -1.5%] | 1     |
| Improvements ✅ <br /> (secondary) | -     | -              | 0     |
| All ❌✅ (primary)                 | -1.5% | [-1.5%, -1.5%] | 1     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=e526d123aa4703aa7b85c5b8e70c20b5ee0deade&end=dd7a12593a5274dd812a7b8ab91a4ff29adb81a2&stat=cycles:u)
This benchmark run did not return any relevant results for this metric.
